From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: chickeneer <emcchickeneer@gmail.com>
Date: Tue, 3 Aug 2021 14:03:09 -0500
Subject: [PATCH] Detect issues with entity removal

Temporary patch to find async entity removals or rekilling dead entities

diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
index b911c8ba875441e686c2a74641dd4e5dcd8a82bc..4e8098328fe36ebd57afbd15f3519aad9ef9e254 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
@@ -751,6 +751,12 @@ public abstract class CraftEntity implements org.bukkit.entity.Entity {
 
     @Override
     public void remove() {
+        if (!org.bukkit.Bukkit.isPrimaryThread()) {
+            new Throwable("Called Async when should be main").printStackTrace();
+        }
+        if (!this.entity.isAlive()) {
+            new Throwable("Entity is already dead...").printStackTrace();
+        }
         this.entity.discard();
     }
 
